libparsec/crates/platform_mountpoint/src/unix/filesystem.rs

```diff
@@ -35,17 +35,29 @@ const GENERATION: u64 = 0;
 const BLOCK_SIZE: u64 = 512 * 1024;
 /// Default permissions for files and folders.
 /// Equivalent to `chmod` flags `all=,u=rwx`.
-const PERMISSIONS: u16 = 0o700;
+const READ_WRITE_PERMISSIONS: u16 = 0o700;
+const READ_ONLY_PERMISSIONS: u16 = 0o500;
 
 fn os_name_to_entry_name(name: &OsStr) -> EntryNameResult<EntryName> {
     name.to_str()
         .map(|s| s.parse())
         .ok_or(EntryNameError::InvalidName)?
 }
 
-fn file_stat_to_file_attr(stat: FileStat, inode: Inode, uid: u32, gid: u32) -> fuser::FileAttr {
+fn file_stat_to_file_attr(
```

Comment:
you can move this function (and `entry_stat_to_file_attr` & `reply_with_lookup`) as `Filesystem` methods if you think it is simpler/cleaner to have `self.is_read_only` (hence not having to modify the call sites) instead of passing it as parameter

Reply:
I don't think we can due to the use of `self.tokio_handle.spawn(async move { ... })`